packages/next/server/app-render.tsx

```diff
@@ -642,13 +642,13 @@ export async function renderToHTMLOrFlight(
     // Only server components can have getServerSideProps / getStaticProps
     // TODO-APP: friendly error with correct stacktrace. Potentially this can be part of the compiler instead.
     if (isClientComponentModule) {
-      if (layoutOrPageMod.getServerSideProps) {
+      if (layoutOrPageMod.ssr) {
         throw new Error(
           'getServerSideProps is not supported on Client Components'
         )
       }
 
-      if (layoutOrPageMod.getStaticProps) {
+      if (layoutOrPageMod.ssg) {
```

Comment:
I don't understand this change, `layoutOrPageMod` should be the client component module from the SSR bundle right? Or the RSC bundle?

Reply:
The checking was for RSC bundle. `createServerComponentRenderer` -> `<ComponentTree />` -> `createComponentTree` -> checking layout module. This is just a temporary workaround, the loader will be reverted to the same version like the one in React and the checking will be moved to build time